README.md

```diff
@@ -427,30 +434,31 @@ values encoded by an extension. This is mostly useful for deduplication and
 optimisation.
 
 ```js
-let symbols = [];
-let decodedSymbols = [];
-
 // preserves ECMAScript Symbol identity and description
-let symbolExtension = {
-  detector: x => typeof x === 'symbol',
-  serialiser(s) {
-    let i = symbols.indexOf(s);
+class SymbolExtension {
+  constructor() {
+    this.symbols = [];
+  }
+  isCandidate(x) {
+    return typeof x === 'symbol' || x && x.constructor === Symbol;
+  }
+  serialise(s) {
+    let i = this.symbols.indexOf(s);
     if (i >= 0) return i;
-    return symbols.push(s) - 1;
-  },
-  deserialiser(n, memo) {
-    if (decodedSymbols[n] == null) {
+    return this.symbols.push(s) - 1;
+  }
+  deserialise(n, memo) {
+    if (this.symbols[n] == null) {
       let s = Symbol(memo[n]);
-      decodedSymbols[n] = s;
+      this.symbols[n] = s;
       return s;
-    } else {
-      return decodedSymbols[n];
     }
-  },
-  memo: () => symbols.map(sym => String(sym).slice(7, -1)),
-};
+    return this.symbols[n];
+  }
+  memo() { return this.symbols.map(sym => String(sym).slice(7, -1)); }
+}
 
-let encoded = encode(data, { extensions: { 0: symbolExtension } });
+let encoded = encode(data, { extensions: { 0: SymbolExtension } });
 ```
 
 Memos are only prepended for extensions that define the optional `memo`
```

Comment:
> Memos are only prepended for extensions that define the optional `memo` function. Memos are encoded and prepended to the SuperPack payload in descending extension point order.

This is an odd mixing of specification requirements ("descending extension point order") with an implementation specific API suggestion ("optional `memo` function").